### PR TITLE
add borrower tests for issue & close

### PR DESCRIPTION
**Pull request by ilinzweilin**
- added additional check for shelf.close()  `require(!nftLocked(loan))`
- moved test user contracts out of test-utils file into sep folder 
- renamed TestUtils contract to TestSetup
- added tests for borrower: issue & close 
- added shared interface file for test contracts. We can add some naming conventions to it.

There are some todos for the close tests:
- will be resolved as soon as refactoring for borrow tests is done

### CODE REVIEW

File: src/core/borrower/shelf.sol [hunk 1]
```diff
@@ -181,7 +179,7 @@ contract Shelf is DSNote, Auth, TitleOwned, Math {
     }
 
     // --- NFT actions ---
-    function lock(uint loan, address usr) public owner(loan) {
+    function lock(uint loan) public owner(loan) {
```
Comment (manuelpolzhofer) [inline src/core/borrower/shelf.sol]: lets say A owns the NFT and the loan. (and he/she added the approve call)

A sends NFT to B. If B would give an approve.

A could still lock the NFT with this method. Correct?

If B would create a new loan with the NFT. Could A still call this method? Would the registry entry still exist?

File: src/core/test/system/borrower/close.t.sol [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity >=0.5.12;
+
+import "../system.sol";
+
+contract CloseTest is SystemTest {
```
Comment (manuelpolzhofer) [inline src/core/test/system/borrower/close.t.sol]: I would prefer renaming the `setUp` method in SystemTest and adding always a `setUp` method in a test.

Which might be just a one liner calling the `defaultSetUp` from System test

File: src/core/test/system/borrower/close.t.sol [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    
+    function closeLoan(uint loanId, uint tokenId, bytes32 lookupId) public {
+        assertPreCondition(loanId, tokenId, lookupId);
+        borrower.close(loanId);
+        assertPostCondition(loanId, tokenId, lookupId);
+    }
```
Comment (manuelpolzhofer) [inline src/core/test/system/borrower/close.t.sol]: 👍  good pattern

File: src/core/test/system/users/borrower.sol [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity >=0.5.12;
+
+import "ds-test/test.sol";
+import { Title } from "tinlake-title/title.sol";
+import "../interfaces.sol";
+
+contract Borrower is DSTest {
```
Comment (manuelpolzhofer) [inline src/core/test/system/users/borrower.sol]:
```suggestion
contract Borrower {
```

If we don't use some log calls it is not required to add it

File: src/core/borrower/shelf.sol (hunk 1 above)
Comment (manuelpolzhofer) [reply src/core/borrower/shelf.sol]: Let's maybe to avoid all the edge case always require the following

```javascript
require(msg.sender == title.ownerOf(loan) == NFTLike(shelf[loan].registry).ownerOf(shelf[loan].tokenId))
```

File: src/core/test/system/users/admin.sol [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity >=0.5.12;
+
+import "ds-test/test.sol";
+import { Title } from "tinlake-title/title.sol";
+import "../interfaces.sol";
+
+contract AdminUser is DSTest{
```
Comment (charleenfei) [inline src/core/test/system/users/admin.sol]: DSTest maybe not needed here

File: src/core/borrower/shelf.sol (hunk 1 above)
Comment (ilinzweilin) [reply src/core/borrower/shelf.sol]: I think it is already given, because the msg.sender has to be the nftOwner and the loanOwner for lock to work.
NFTLike(shelf[loan].registry).transferFrom(msg.sender, address(this), shelf[loan].tokenId); -> makes sure user is nftOwner
modifier: owner(loan) -> makes sure user is loanOwner

File: src/core/test/system/users/borrower.sol (hunk 4 above)
Comment (ilinzweilin) [reply src/core/test/system/users/borrower.sol]: will remove after finishing developing all the tests.

File: src/core/test/system/borrower/close.t.sol (hunk 2 above)
Comment (ilinzweilin) [reply src/core/test/system/borrower/close.t.sol]: will rename in next PR
